src/commonControls/MultiValueListControl/MultiValueListControlAPL.ts

```diff
@@ -0,0 +1,249 @@
+                {
+                    name: 'alexa-layouts',
+                    version: '1.2.0',
+                },
+            ],
+            layouts: {},
+            mainTemplate: {
+                parameters: ['payload'],
+                items: [
+                    {
+                        type: 'Container',
+                        id: 'root',
+                        bind: [
+                            {
+                                name: 'debugText',
+                                type: 'string',
+                                value: 'debug...',
+                            },
+                            {
+                                name: 'showDebug',
+                                type: 'boolean',
+                                value: false,
+                            },
+                        ],
+                        items: [
+                            {
+                                type: 'AlexaBackground',
+                            },
+                            {
+                                type: 'AlexaHeader',
+                                headerDivider: true,
+                                headerTitle: '${payload.general.headerTitle}',
+                                headerSubtitle: '${payload.general.headerSubtitle}',
+                                height: '20vh',
+                            },
+                            {
+                                type: 'AlexaButton',
+                                buttonText: 'Done',
+                                id: 'actionComplete',
+                                primaryAction: {
+                                    type: 'SendEvent',
+                                    arguments: ['${payload.general.controlId}', 'Complete'],
+                                },
+                                right: '10vw',
+                                position: 'absolute',
+                            },
+                            {
+                                type: 'Text',
+                                id: 'DebugText',
+                                text: '${debugText}',
+                                display: "${showDebug ? 'normal' : 'invisible'}",
+                                position: 'absolute',
+                                right: '0vw',
+                            },
+                            {
+                                type: 'Container',
+                                paddingLeft: '@spacingMedium',
+                                direction: 'row',
+                                items: [
+                                    {
+                                        type: 'AlexaTextList',
+                                        width: '50vw',
+                                        height: '80vh',
+                                        touchForward: true,
+                                        headerBackButton: false,
+                                        primaryAction: {
```

Comment:
suggest trialing "disable" functionality so that user can't press more buttons before the round trip completes and cause race conditions.
Although, if the operations are all safe to race then it might be a feature.

I'm concerned about remove which is currently implemented via Ordinals which are race-sensitive.  If removal changed to being by (value, ordinal), then races might be OK.  The server logic would be to remove the first matching item at ordinal or lower to allow for races such as "delete item #1" followed immediately by "delete item #2" which would need to be processed as a second instance of delete item#1.

tl;dr:   avoiding races is safest but in this particular case you might be able to make races safe and could be a nice feature?

Reply:
As discussed, will set `disabled: boolean` property on list of choices when selected and also will disable swipe action on items when a single item is removed to avoid race conditions.